src/LEGO.AsyncAPI.Readers/JsonHelper.cs

```diff
@@ -4,15 +4,49 @@ namespace LEGO.AsyncAPI.Readers
 {
     using System;
     using System.Globalization;
+    using System.IO;
+    using System.Text.Encodings.Web;
+    using System.Text.Json;
     using System.Text.Json.Nodes;
     using LEGO.AsyncAPI.Exceptions;
 
+    /// <summary>
+    /// Contains helper methods for working with Json 
+    /// </summary>
     internal static class JsonHelper
     {
-        public static string GetScalarValue(this JsonNode node)
+        private static readonly JsonWriterOptions WriterOptions;
+
+        static JsonHelper()
+        {
+            WriterOptions = new JsonWriterOptions()
+            {
+                Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping,
+                Indented = false,
+                MaxDepth = 1,
+                SkipValidation = true,
+            };
+        }
+
+        /// <summary>
+        /// Takes a <see cref="JsonValue"/> and converts it into a string value.
+        /// </summary>
+        /// <param name="jsonValue">The node to convert.</param>
+        /// <returns>The string value.</returns>
+        public static string GetScalarValue(this JsonValue jsonValue)
         {
-            var scalarNode = node is JsonValue value ? value : throw new AsyncApiException($"Expected scalar value");
-            return Convert.ToString(scalarNode.GetValue<object>(), CultureInfo.InvariantCulture);
+            using (MemoryStream memoryStream = new MemoryStream())
+            using (Utf8JsonWriter writer = new Utf8JsonWriter(memoryStream, WriterOptions))
+            {
+                jsonValue.WriteTo(writer);
+                writer.Flush();
+                memoryStream.Position = 0;
+                using (StreamReader reader = new StreamReader(memoryStream))
+                {
+                    string value = reader.ReadToEnd();
+                    return value.Trim('"');
```

Comment:
The old method would not handle conversation of DateTime's correctly which was causing it to fail. Instead now we use the built in logic to serialization of values and then just remove the contain quotes.